src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -1857,15 +1871,22 @@ def forward(
 
         itc_loss = None
 
-        if labels is not None:
-            labels = torch.arange(len(labels), device=logits.device)
+        if return_loss:
+            labels = torch.arange(len(logits), device=logits.device)
             text_to_image_loss = nn.functional.cross_entropy(logits_text_to_image, labels)
             text_to_cross_loss = nn.functional.cross_entropy(logits_text_to_cross, labels)
             image_to_cross_loss = nn.functional.cross_entropy(logits_image_to_cross, labels)
             itc_loss = (text_to_image_loss + text_to_cross_loss + image_to_cross_loss) / 3.0
 
         if not return_dict:
-            output = tuple(logits)
+            output = (logits, text_embeds, image_embeds, cross_embeds)
+            if output_attentions:
+                output = output + (outputs[4],)
+                if output_hidden_states:
+                    output = output + (outputs[3],)
+            elif output_hidden_states:
+                output = output + (outputs[3],)
```

Comment:
I am wondering why `BridgeTowerModelOutput` and `BridgeTowerContrastiveOutput` will have `attentions` and `hidden_states` swapped, and causing the somehow strange order here.

IMO, it's much better to have a consistent ordering. I would also like  hear @amyeroberts 's opinion.

Reply:
Also, I can't comment at the exact line , but the block below  is better to put the arguments in the order that is defined in `BridgeTowerContrastiveOutput`. In terms of functionality, it is fine, but putting in the same order will reduce some (mental) burden.

```python
        return BridgeTowerContrastiveOutput(
            attentions=outputs.attentions,
            hidden_states=outputs.hidden_states,
            text_embeds=text_embeds,
            image_embeds=image_embeds,
            cross_embeds=cross_embeds,
            logits=logits,
            loss=itc_loss,
        )
```

Reply:
Yes, I agree, it's better to have consistent ordering. I think most, if not all, other model outputs return hidden_states then attentions

Reply:
We fixed this in the latest commit. Thanks for suggestion.